### PR TITLE
Tim Estimation, Span and Settings

### PR DESCRIPTION
**Pull request by X-Raym**
- Colors settings for Progress Bar
- Time evaluation based on character per seconds settings
- Re-ordering of settings variable
- Show toggle for all fields
- Span class for all fields to allow custom style

My only concern is that I renamed the hidechars variable to showchars for consistency, I wonder how Atom handle settings variables change.

### CODE REVIEW

File: lib/main.coffee [hunk 1]
```diff
@@ -37,10 +37,14 @@ module.exports =
...
-      description: 'Use a CSS color value, such as rgb(0, 85, 255) or green'
-      type: 'string'
+      type: 'color'
       default: 'rgb(0, 85, 0)'
       order: 5
+    goalBgColor:
+      title: 'Color for word goal background'
+      type: 'color'
+      default: 'red'
+      order: 6
```
Comment (OleMchls) [inline lib/main.coffee]: By introducing the `6` here all following options would need to be updated.

**Comment by X-Raym** [comment]
@OleMchls As I suspected, there is a problem with updating settings variable names.
If one boolean was set to non default, , the settings is saved in Atom files,
so if you change the variable settings name of a package,
this obsolete setting will appear at the end of the settings package.

Here is a demo, where I first had the old hidechars settings set to true (non default), and when I updated with the new settings (called showchars):

Alt text

As you can see, there is this obsolete value.

 I assume changing settings variable name is something common with package updates, but I don't find how to either Hide this old variable or how to delete properly. Any Idea ?

**Comment by OleMchls** [comment]
What about the options that a user has set, where we rename the option name? Will these settings be converted? While this is not a deal breaker for me, I think we'd need to make this a major release.

**Comment by X-Raym** [comment]
@OleMchls In this case it fallback to default value. Fortunately, it is just a toggle to do in the settings, and only for those who use the hidechars. I don't think this simple toggle for only a part of the users deserve to be a 'major' release 😄 But sure, backward compatibility is otfen. It is always a dilemma with Refactorization and Backward compatibility. If backward compatibility is more important, we can use the old hidechars variable and treat it as if it was a showchars boolean.

**Comment by OleMchls** [comment]
It's released as `3.0.0` thank you for the contribution!

**Comment by X-Raym** [comment]
@OleMchls awesome ! Many thanks for this, it's an honor to contribute :P

**Pull request merged**